flow-decomposition/src/main/java/com/powsybl/flow_decomposition/FlowDecompositionComputer.java

```diff
@@ -67,19 +69,50 @@ public FlowDecompositionResults run(Network network) {
         return flowDecompositionResults;
     }
 
+    private List<Branch> getXnecList(Network network, Map<String, Map<Country, Double>> zonalPtdf) {
+        XnecSelector xnecSelector;
+        switch (parameters.getXnecSelectionStrategy()) {
+            case ONLY_INTERCONNECTIONS:
+                xnecSelector = new XnecSelectorInterconnection();
+                break;
+            case ZONE_TO_ZONE_PTDF_CRITERIA:
+                xnecSelector = new XnecSelector5percPtdf(zonalPtdf);
+                break;
+            default:
+                throw new PowsyblException(String.format("XnecSelectionStrategy %s is not valid",
+                    parameters.getXnecSelectionStrategy()));
+        }
+        return xnecSelector.run(network);
+    }
+
     private static LoadFlowParameters initLoadFlowParameters() {
         LoadFlowParameters parameters = LoadFlowParameters.load();
         parameters.setDc(DC_LOAD_FLOW);
         LOGGER.debug("Using following load flow parameters: {}", parameters);
         return parameters;
     }
 
+    private Map<Country, Map<String, Double>> getGlsks(Network network,
+                                                       FlowDecompositionResults flowDecompositionResults) {
+        GlskComputer glskComputer = new GlskComputer();
+        Map<Country, Map<String, Double>> glsks = glskComputer.run(network);
+        return flowDecompositionResults.saveGlsks(glsks);
+    }
+
+    private Map<String, Map<Country, Double>> getZonalPtdf(Network network,
+                                                           Map<Country, Map<String, Double>> glsks,
+                                                           FlowDecompositionResults flowDecompositionResults) {
+        ZonalSensitivityAnalyser zonalSensitivityAnalyser = new ZonalSensitivityAnalyser(loadFlowParameters, parameters);
+        Map<String, Map<Country, Double>> zonalPtdf = zonalSensitivityAnalyser.run(network,
+            glsks, SensitivityVariableType.INJECTION_ACTIVE_POWER);
+        return flowDecompositionResults.saveZonalPtdf(zonalPtdf);
+    }
+
     private Map<Country, Double> getZonesNetPosition(Network network,
                                                      FlowDecompositionResults flowDecompositionResults) {
         NetPositionComputer netPositionComputer = new NetPositionComputer(loadFlowParameters);
         Map<Country, Double> netPosition = netPositionComputer.run(network);
-        flowDecompositionResults.saveACNetPosition(netPosition);
-        return netPosition;
+        return flowDecompositionResults.saveACNetPosition(netPosition);
```

Comment:
I'm not a fan of this change. Returning the value as is from the "save" methods only makes the code harder to read, especially since they are not modified at all.
I'd prefer the "save" methods continue to return void, and return what you need to return explicitly here (like before).

Reply:
done